acceptance/tests/provider/package/rpm_ensure_install_multiversion_package.rb

```diff
@@ -0,0 +1,66 @@
+test_name "dpkg ensure held package should preserve version if package is allready installed"
+confine :to, :platform => /redhat|centos|el|fedora/
+tag 'audit:low'
+
+require 'puppet/acceptance/common_utils'
+extend Puppet::Acceptance::PackageUtils
+extend Puppet::Acceptance::ManifestUtils
+
+package = "kernel-devel"
+
+agents.each do |agent|
+  initially_installed_versions = []
+
+  teardown do
+    available_versions = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Available Packages/ d' | tr -s ' ' | cut -d ' ' -f 2").stdout
+    initially_installed_versions.each{ |v|
+      if available_versions.include? v
+        package_manifest = resource_manifest('package', package, ensure: v)
+        apply_manifest_on(agent, package_manifest, :catch_failures => true)
+      end
+    }
+  end
+
+  step "Save initially installed package versions" do
+    installed_versions = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Installed Packages/ d' -e '/Available Packages/,$ d' | tr -s ' ' | cut -d ' ' -f 2").stdout.split("\n")
+    available_versions = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Available Packages/ d' | tr -s ' ' | cut -d ' ' -f 2").stdout.split("\n")
+    if installed_versions.size + available_versions.size < 2
+      skip_test "we need at least two package versions to perform the multiverion rpm test"
+    end
+    initially_installed_versions = installed_versions
+  end
+
+  step "Uninstall package versions for clean setup" do
+    package_manifest = resource_manifest('package', package, ensure: 'absent')
+    apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+      remaining_installed_versions = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Installed Packages/ d' -e '/Available Packages/,$ d' | tr -s ' ' | cut -d ' ' -f 2").stdout
+      assert(remaining_installed_versions.empty?)
+    end
+  end
+
+  step "Ensure oldest version of multiversion package is installed" do
+    oldest_version = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Available Packages/ d' | head -1 | tr -s ' ' | cut -d ' ' -f 2").stdout.strip!
+    package_manifest = resource_manifest('package', package, ensure: oldest_version)
+    apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+      installed_version = on(agent.name, "rpm -q #{package}").stdout
+      assert_match(/#{oldest_version}/, installed_version)
+    end
+  end
+
+  step "Ensure newest package multiversion package in installed" do
+    newest_version = on(agent.name, "yum --showduplicates list kernel-devel | sed -e '1,/Available Packages/ d' | tail -1 | tr -s ' ' | cut -d ' ' -f 2").stdout.strip!
```

Comment:
I think `String#strip` would work here since you don't need to change the string in place (?)